### ISSUE TITLE
Prefill Processing

### ISSUE THREAD

**Issue by Nick-infinity** [karpathy/llama2.c#445]
Hello, I am sorry if question is very basic but need a little help over here.

Cant we just skip the attention processing and continue from here for input prompt tokens. 
The KV Cache will be built for all the input tokens.
Why do we need to calculate the attention for all the input tokens as well down till logits?
I cant seem to find the connection between the input tokens and generated tokens apart from the KV cache.

https://github.com/karpathy/llama2.c/blob/d9862069e7ef665fe6309e3c17398ded2f121bf5/run.c#L280

@karpathy

**Comment by Nick-infinity** [comment]
the X activation is fed back for multiple layers after being changed with help of attention output. and that's why attention is needed I guess for input tokens as well.

**Comment by Nick-infinity** [comment]
We can definitely avoid
`  rmsnorm(x, x, w->rms_final_weight, dim);

    // classifier into logits
    matmul(s->logits, x, w->wcls, p->dim, p->vocab_size);`

for input tokens . It will give some perf bump
